### PR TITLE
Bug 1156635 - [Stingray][Contribution] Browser

### PR DESCRIPTION
**Pull request by begeeben**
Merge partner code contribution
- Modify unsupported api calls and make it work on B2G.
- Remove ifilter.
- Remove duplicated shared lib.
- Remove proprietary tvstore.
- Remove isSetupLockActive.
- Remove mozTV, inputPortManager and panaInputDeviceSetting.
- Remove ASR.
- Remove preventDefaultForVideo.
- Remove launch_from property.
- Remove meta-inf folder.
- Change default screen mode to full screen.
- Set default home screen and screen shot to mozilla.org.
- Revise global l10n reference.
- Remove browser widget.
- Remove remote.js.
- Correct linting errors.
- Clean up manifest.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e5cff30ecf872de8053e5b31222e2f4fcacf8c3c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=89d97a2cfe1c4424707e05f3f263fec6a6e104da

**Comment by rexboy7** [comment]
Please add a `README.md` to describe that this app is modified from partner's contribution. (With repository address, if possible)

File: tv_apps/browser/manifest.webapp [hunk 1]
```diff
@@ -1,61 +1,210 @@
...
     }
   },
   "default_locale": "en-US",
   "icons": {
-    "84": "/shared/resources/branding/browser_84.png",
-    "126": "/shared/resources/branding/browser_126.png",
-    "142": "/shared/resources/branding/browser_142.png",
-    "189": "/shared/resources/branding/browser_189.png",
-    "284": "/shared/resources/branding/browser_284.png"
+    "336": "/style/icons/icon.png"
```
Comment (rexboy7) [inline tv_apps/browser/manifest.webapp]: Let's use the original `/shared/resources/branding/..`

File: tv_apps/browser/index.html [hunk 2]
```diff
@@ -1,364 +1,742 @@
...
-    <link rel="localization" href="locales/browser.{locale}.properties">
+    <meta name="availableLanguages" content="de-DE:201506220909, es-ES:201506220909, nl-NL:201506220909, en-IE:201506220909, en-GB:201506220909, no-NO:201506220909, fi-FI:201506220909, fr-FR:201506220909, da-DK:201506220909, tr-TR:201506220909, it-IT:201506220909, sv-SE:201506220909, pt-PT:201506220909, el-GR:201506220909, hu-HU:201506220909, hr-HR:201506220909, cs-CZ:201506220909, ru-RU:201506220909, bg-BG:201506220909, ro-RO:201506220909, sk-SK:201506220909, sl-SI:201506220909, pl-PL:201506220909, sr-YU:201506220909, et-EE:201506220909, lv-LV:201506220909, lt-LT:201506220909, en-AU:201506220909, en-HK:201506220909, zh-CN:201506220909, ar:201506220909, fa:201506220909, th:201506220909, vi:201506220909, id:201506220909, zh-HK:201506220909, en-BR:201506220909, pt-BR:201506220909, es-BR:201506220909, hi:201506220909, zh-TW:201506220909, en-TW:201506220909, ja-JP:201506220909, en-US:201506220909, es-US:201506220909, fr-CA:201506220909, ru:201506220909, he:201506220909, my:201506220909, mk-MK:201506220909">
+    <link rel="manifest" href="manifest.webapp">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: Seems this line is no longer needed. (see bug 1115807)

File: tv_apps/browser/index.html [hunk 3]
```diff
@@ -1,364 +1,742 @@
...
-    <link rel="localization" href="/shared/locales/branding/branding.{locale}.properties">
-    <link rel="localization" href="locales/browser.{locale}.properties">
+    <meta name="availableLanguages" content="de-DE:201506220909, es-ES:201506220909, nl-NL:201506220909, en-IE:201506220909, en-GB:201506220909, no-NO:201506220909, fi-FI:201506220909, fr-FR:201506220909, da-DK:201506220909, tr-TR:201506220909, it-IT:201506220909, sv-SE:201506220909, pt-PT:201506220909, el-GR:201506220909, hu-HU:201506220909, hr-HR:201506220909, cs-CZ:201506220909, ru-RU:201506220909, bg-BG:201506220909, ro-RO:201506220909, sk-SK:201506220909, sl-SI:201506220909, pl-PL:201506220909, sr-YU:201506220909, et-EE:201506220909, lv-LV:201506220909, lt-LT:201506220909, en-AU:201506220909, en-HK:201506220909, zh-CN:201506220909, ar:201506220909, fa:201506220909, th:201506220909, vi:201506220909, id:201506220909, zh-HK:201506220909, en-BR:201506220909, pt-BR:201506220909, es-BR:201506220909, hi:201506220909, zh-TW:201506220909, en-TW:201506220909, ja-JP:201506220909, en-US:201506220909, es-US:201506220909, fr-CA:201506220909, ru:201506220909, he:201506220909, my:201506220909, mk-MK:201506220909">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: Seems we only need to keep `en-US.properties` and `content="en-US"`.
(https://bugzilla.mozilla.org/show_bug.cgi?id=1099321)

File: tv_apps/browser/index.html [hunk 4]
```diff
@@ -1,364 +1,742 @@
...
+    <link rel="manifest" href="manifest.webapp">
+    <link rel="localization" href="/locales-obj/{locale}.json">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: We may need to use `/locales/properties.json`

File: tv_apps/browser/manifest.webapp [hunk 5]
```diff
@@ -1,61 +1,210 @@
...
     "systemXHR":{},
-    "settings":{ "access": "readwrite" },
-    "geolocation" : {},
-    "desktop-notification" : {},
-    "audio-capture": {},
-    "video-capture": {},
-    "device-storage:pictures" : { "access": "readwrite" },
-    "device-storage:videos" : { "access": "readwrite" },
-    "device-storage:music" : { "access": "readwrite" }
+    "storage": {},
```
Comment (rexboy7) [inline tv_apps/browser/manifest.webapp]: This permission is no longer functional.
(bug 1126638)

File: tv_apps/browser/index.html [hunk 6]
```diff
@@ -1,364 +1,742 @@
...
-                 data-l10n-id="top-sites-tab" role="tab"
-                 aria-selected="true" aria-controls="top-sites-panel">Top Sites</a>
+    <!-- main -->
+    <div id="main-block" data-mode="cursor" class="main-window">
+      <!-- toolbar -->
+      <ul id="toolbar-panel" data-pvt="false">
+        <div id="private-bar"></div>
+        <li id="sidebar-button-block" data-tips="WB_LT_TIPS_CHANGE_SCREEN" data-fade="false">
+          <div id="sidebar-button" data-sidebar="true">
+            <svg id="sidebar-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid slice">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: Let's add `XXX: Replace the SVG to icon font` for each SVGs embedded inside index.html

File: tv_apps/browser/index.html [hunk 7]
```diff
@@ -1,364 +1,742 @@
...
+                   M37.129,21.849v0.955H23.839v-4.222h10.472C35.895,18.583,37.129,20.266,37.129,21.849z M34.312,35.419H23.839V24.905h13.289
+                  v7.834C37.129,34.322,35.895,35.419,34.312,35.419z"></path>
+              </g>
+            </svg>
+          </div>
+        </li>
+
+        <li id="back-button-block" data-tips="WB_LT_TIPS_HISTORY_BACK">
+          <div id="back-button">
+            <svg id="back-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 8]
```diff
@@ -1,364 +1,742 @@
...
+              <path class="back-button-path" fill-rule="evenodd" clip-rule="evenodd" fill="#1D1D1B" d="M39.325,24.619H21.352l6.701-6.701
+              c-0.38-1.648-0.861-3.185-1.428-4.584L15.46,24.5c-1.385,1.384-1.385,3.628,0,5.012l11.153,11.154
+              c0.567-1.396,1.048-2.928,1.431-4.57L21.349,29.4h17.976C39.662,27.764,39.661,26.256,39.325,24.619z"></path>
+            </svg>
+          </div>
+        </li>
+        <li id="forward-button-block" class="disable" data-tips="WB_LT_TIPS_HISTORY_FORWARD">
+          <div id="forward-button-border">
+            <div id="forward-button">
+              <svg id="forward-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 9]
```diff
@@ -1,364 +1,742 @@
...
+          <div></div>
+        </li>
+        <li id="addressbar-block">
+          <form id="address-form">
+            <input id="private-icon" value="" type="image">
+            <input id="ssl-indicator" value="" type="image">
+            <input id="url-input" dir="auto" type="text">
+            <div class="input-line"></div>
+            <div id="address-button" data-mode="load" data-tips="WB_LT_TIPS_RELOAD">
+              <svg id="address-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 10]
```diff
@@ -1,364 +1,742 @@
...
+          </form>
+        </li>
+
+        <li id="searchbar-block">
+          <form id="search-form">
+            <input id="search-favicon" value="" type="image">
+            <input id="search-input" data-tips="WB_LT_TIPS_SEARCH_BAR" dir="auto" type="text">
+            <div class="input-line"></div>
+            <div id="search-button">
+              <svg id="search-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 11]
```diff
@@ -1,364 +1,742 @@
...
+            <div id="search-clear-button"></div>
+          </form>
+          <button id="search-close-button" data-l10n-id="LT_NETWORK_STATUS_BTN_CLOSE">close</button>
+        </li>
+
+        <li id="iconbutton-block">
+          <ul>
+            <li id="bookmark-button-block" data-tips="WB_LT_TIPS_BOOKMARK_THIS_PAGE">
+              <div id="bookmark-button" data-isbookmark="false">
+                <svg id="bookmark-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 12]
```diff
@@ -1,364 +1,742 @@
...
+                        27,35.917 36.96,41.327 34.893,30.183 			"></polygon>
+                      </g>
+                    </g>
+                  </g>
+                </svg>
+              </div>
+            </li>
+            <li id="show-bookmarks-button-block" data-tips="WB_LT_TIPS_SHOW_YOUR_BOOKMARK">
+              <div id="show-bookmarks-button">
+                <svg id="show-bookmarks-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 13]
```diff
@@ -1,364 +1,742 @@
...
+                  c-1.205,0-2.181,0.977-2.181,2.181v18.172c0,1.204,0.976,2.181,2.181,2.181h21.102c1.204,0,2.181-0.977,2.181-2.181V20.617
+                  C40.731,19.413,39.755,18.436,38.551,18.436z M21.83,38.06h-4.363v-4.363h4.363V38.06z M21.833,31.521h-4.361v-4.361h4.361
+                  V31.521z M21.833,24.978h-4.361v-4.361h4.361V24.978z M23.991,27.159h14.56v4.361h-14.56V27.159z M38.552,38.06H23.999v-4.363
+                  h14.553V38.06z M38.542,24.968l-14.538,0.007v-4.356h14.548L38.542,24.968z"></path>
+                </svg>
+              </div>
+            </li>
+            <li id="home-button-block" data-tips="WB_LT_TIPS_HOMEPAGE">
+              <div id="home-button">
+                <svg id="home-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 14]
```diff
@@ -1,364 +1,742 @@
...
+                </svg>
+              </div>
             </li>
-            <li role="presentation">
-              <a id="bookmarks-tab" href="#"
-                 data-l10n-id="bookmarks" role="tab"
-                 aria-selected="false" aria-controls="bookmarks-panel">Bookmarks</a>
+            <li id="zoom-button-block" data-tips="WB_LT_TIPS_ZOOM_IN" data-colorbar="red">
+              <div id="zoom-button">
+                <svg id="zoom-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 15]
```diff
@@ -1,364 +1,742 @@
...
+                </svg>
+              </div>
             </li>
-            <li role="presentation">
-              <a id="history-tab" href="#"
-                 data-l10n-id="history" role="tab"
-                 aria-selected="false" aria-controls="history-panel">History</a>
+            <li id="tabs-button-block" data-tips="WB_LT_TIPS_TABS" data-tabs="tab01" data-colorbar="yellow">
+              <div id="tabs-button">
+                <svg id="tabs-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 16]
```diff
@@ -1,364 +1,742 @@
...
+                  c1.968,0,3.502-1.364,3.502-3.332V22.656C40.315,20.688,38.782,18.595,36.814,18.595z M35.394,30.061
+                  c0,0.499-0.405,1.127-0.904,1.127h-3.981v4.006c0,0.499-0.602,0.904-1.102,0.904h-0.524c-0.5,0-1.189-0.405-1.189-0.904v-4.006
+                  h-4c-0.5,0-0.904-0.628-0.904-1.127v-0.452c0-0.5,0.405-1.232,0.904-1.232h4v-3.982c0-0.5,0.689-0.904,1.189-0.904h0.524
+                  c0.5,0,1.102,0.404,1.102,0.904v3.982h3.981c0.5,0,0.904,0.732,0.904,1.232V30.061z"></path>
+                </svg>
+            </div>
+          </li>
+            <li id="menu-button-block" data-tips="WB_LT_TIPS_MENU">
+              <div id="menu-button">
+                <svg id="menu-button-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="3 3 47 47" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 17]
```diff
@@ -1,364 +1,742 @@
...
-              <li class="top-site-item">
-                <a href="http://mozilla.org"></a>
-                <span>Mozilla</span>
-              </li>
+        </li>
+      </ul>
+
+      <!-- bookmark animation -->
+      <div id="bookmark-button-anime">
+        <svg id="bookmark-button-anime-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="0 0 54 54" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/index.html [hunk 18]
```diff
@@ -1,364 +1,742 @@
...
+          <polygon class="bookmark-button-path" fill="#00CAF2" points="43.116,22.382 31.878,20.904 27,10.673 22.122,20.904 10.884,22.382 19.107,30.183 17.04,41.327
+          27,35.917 36.96,41.327 34.893,30.183"></polygon>
+        </svg>
+      </div>
+      <div id="show-bookmarks-button-anime">
+        <svg id="show-bookmarks-button-anime-svg" xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="0 0 54 54" x="0" y="0" preserveAspectRatio="xMidyMid meet">
```
Comment (rexboy7) [inline tv_apps/browser/index.html]: same above

File: tv_apps/browser/js/browser_db.js [hunk 19]
```diff
@@ -55,6 +40,7 @@ var BrowserDB = {
    * Populate topsites data from mozSettings to database.
    */
   initSingleVariant: function browserDB_initSingleVariant() {
+/* TODO: mozSettings to indexedDB
```
Comment (rexboy7) [inline tv_apps/browser/js/browser_db.js]: Seems they are preventing initializing browserDB. Will we restore it or just remove it?

Comment (rexboy7) [inline tv_apps/browser/js/browser_db.js]: If we are going to remove it, please just try to remove unused codes. If not and we don't want them to activate for now, please add some XXX to describe why we need to keep these codes.

File: tv_apps/browser/js/search_util.js [hunk 20]
```diff
@@ -0,0 +1,283 @@
...
+  /* URL of each search engine */
+  SEARCH_URL: {
+    //Google :
+    0 : [
+      "http://www.google.DOMAIN/search?ie=UTF-8&oe=UTF-8&q=",
+      "http://www.google.DOMAIN/search?ie=UTF-8&oe=UTF-8&tbm=isch&q=",
+      "http://www.google.DOMAIN/search?ie=UTF-8&oe=UTF-8&tbm=vid&q=",
+      "http://www.google.DOMAIN/search?ie=UTF-8&oe=UTF-8&tbm=nws&q=",
+      "http://maps.google.com/maps?q=",
+      "http://www.youtube.DOMAIN/results?search_query="
```
Comment (rexboy7) [inline tv_apps/browser/js/search_util.js]: These search URLs may need to be confirmed (and remove inappropriate one) before check-in.

File: tv_apps/browser/js/search_util.js [hunk 21]
```diff
@@ -0,0 +1,283 @@
...
+      [ [ "www.google.",  "/search", ""         ] ],
+      [ [ "www.google.",  "/search", "tbm=isch" ],
+        [ "www.google.",  "/imghp",  ""         ] ],
+      [ [ "www.google.",  "/search", "tbm=vid"  ],
+        [ "www.google.",  "/videohp",""         ] ],
+      [ [ "www.google.",  "/search", "tbm=nws"  ],
+        [ "news.google.", "/nwshp",  ""         ] ],
+      [ [ "maps.google.", "",        ""         ],
+        [ "www.google.",  "/maps",   ""         ] ],
+      [ [ "www.youtube.", "",        ""         ] ]
```
Comment (rexboy7) [inline tv_apps/browser/js/search_util.js]: Same above.

File: tv_apps/browser/locales-obj/ar.json [hunk 22]
```diff
@@ -0,0 +1 @@
..._HOMEPAGE","$v":"ضبط كصفحة رئيسية"},{"$i":"WB_LT_SET_HOMEPAGE","$v":["تم ضبط ",{"t":"idOrVar","v":"value0"}," الآن كصفحتك الرئيسية"]},{"$i":"WB_LT_SWITCH_CURSOR_MODE","$v":"لقد قمت بالتحويل إلى وضع المؤشر"},{"$i":"WB_LT_SWITCH_PAN_MODE","$v":"لقد قمت بالتحويل إلى وضع السحب"},{"$i":"WB_LT_TIPS_BOOKMARK_THIS_PAGE","$v":"إضافة هذه الصفحة للتفضيلات"},{"$i":"WB_LT_TIPS_CHANGE_SCREEN","$v":"تغيير الشاشة"},{"$i":"WB_LT_TIPS_HISTORY_BACK","$v":"الرجوع إلى السجل الزمني"},{"$i":"WB_LT_TIPS_HISTORY_FORWARD","$v":"تقديم السجل الزمني"},{"$i":"WB_LT_TIPS_HOMEPAGE","$v":"عرض الصفحة الرئيسية"},{"$i":"WB_LT_TIPS_MENU","$v":"القائمة"},{"$i":"WB_LT_TIPS_NEW_PRIVATE_WINDOW","$v":"نافذة خاصة جديدة"},{"$i":"WB_LT_TIPS_OPEN_A_NEW_TAB","$v":"فتح علامة تبويب جديدة"},{"$i":"WB_LT_TIPS_RELOAD","$v":"إعادة التحميل"},{"$i":"WB_LT_TIPS_SEARCH_BAR","$v":"قم بالبحث على الويب"},{"$i":"WB_LT_TIPS_SHOW_YOUR_BOOKMARK","$v":"عرض التفضيلات"},{"$i":"WB_LT_TIPS_STOP_LOADING","$v":"إيقاف فتح هذه الصفحة"},{"$i":"WB_LT_TIPS_S_SEARCH","$v":[{"t":"idOrVar","v":"value0"}," بحث"]},{"$i":"WB_LT_TIPS_TABS","$v":"عرض علامات التبويب"},{"$i":"WB_LT_TIPS_ZOOM_IN","$v":"تكبير"},{"$i":"WB_LT_W_BOOKMARK_DELETE","$v":"حذف علامة الموقع"}]
```
Comment (rexboy7) [inline tv_apps/browser/locales-obj/ar.json]: We may need to remove files under `tv_apps/browser/locales-obj/` and use locales/en-US.properties instead.

File: tv_apps/browser/manifest.webapp [hunk 23]
```diff
@@ -1,61 +1,210 @@
...
+    "desktop-notification" : {}
   },
   "locales": {
     "ar": {
-      "name": "المتصفح",
-      "description": "Gaia متصفح الويب"
+      "name": "Browser",
```
Comment (rexboy7) [inline tv_apps/browser/manifest.webapp]: Not sure if we need so many locales here... Maybe we just need en-US and some of them.

File: tv_apps/browser/manifest.webapp (hunk 1 above)
Comment (begeeben) [inline tv_apps/browser/manifest.webapp]: Thanks! Use 284 pixel icon temporarily. Track this in Bug 1196624 - Need official and unofficial 336 pixel Firefox branding icon for Firefox OS TV .

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c766ed7f552da315b0f57a9a2f5e989cc43db0d4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ddd769cda1b804c728cd24a2db67561e6c48217a

**Comment by rexboy7** [comment]
We may remove the icon.png

File: tv_apps/browser/js/browser_db.js [hunk 24]
```diff
@@ -99,33 +89,33 @@ var BrowserDB = {
...
-          return;
-        }
-
-        this.setAndLoadIconForPage(topSite.uri, topSite.iconUri);
-      }
-    }, this);
-  },
+  //populateTopSites: function browserDB_populateTopSites(data, frequency) {
+  //  var index = frequency;
+  //  console.log("### populateTopsite data.iconUri=" + data.iconUri);
```
Comment (rexboy7) [inline tv_apps/browser/js/browser_db.js]: remove these console.log?

Comment (begeeben) [inline tv_apps/browser/js/browser_db.js]: I suggest we preserve these commented lines. We could decide if we would like to bring back the original v2.0 topsite display logic later.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5b2304b2187dec6d29cfd012d8c800f69678ce50

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5feeab0e23392c5361a1490ce31a11ab84c2ccf6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=97d99432fd667e9c40f99188ed1a7c253367b0cf

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a39780a561781765229d4035f8fb063e6b866a27

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=32496b961d904fbfa3612119790bf98040dbd31a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a11b64bd063db20da48329e0c6f47535d311cd46

File: tv_apps/browser/js/authentication_dialog.js [hunk 25]
```diff
@@ -0,0 +1,410 @@
...
+      //this.httpAuthenticationMessage.setAttribute('data-l10n-id',
+        //'the-username-or-password-is-incorrect');
+      //this.httpAuthenticationMessage.classList.add('error');
+    } else {
+      var msg = navigator.mozL10n.get('WB_LT_AUTH_MESSAGE_1_1')
+              + ' ' + evt.detail.host + ' '
+              + navigator.mozL10n.get('WB_LT_AUTH_MESSAGE_1_2') + '<br>'
+              + navigator.mozL10n.get('WB_LT_AUTH_MESSAGE_2') + ' '
+              + evt.detail.realm;
+      this.httpAuthenticationMessage.innerHTML = msg.trim();
```
Comment (zbraniecki) [inline tv_apps/browser/js/authentication_dialog.js]: This should not look like this. Instead it should be sth like:

`WB_LT_AUTH_MESSAGE = The server {{ host }} requires a user name and password.<br/>Message from the server {{ msg }}`

and in JS:

`navigator.mozL10n.setAttributes(this.httpAuthenticationMessage, 'WB_LT_AUTH_MESSAGE', {
  host: evt.detailt.host,
  msg: evt.detail.realm
});`

File: tv_apps/browser/js/browser_dialog.js [hunk 26]
```diff
@@ -0,0 +1,554 @@
...
+    this.dialogEvt = evt;
+    this.browserDialogBase.style.display = 'block';
+
+    switch(type) {
+      case 'del_cookie':
+        opt = {
+          title: null,
+          msg: _('WB_LT_CONFIRM_DELETE_COOKIES'),
+          bt1: _('LT_CANCEL'),
+          bt2: _('WB_LT_CLEAR')
```
Comment (zbraniecki) [inline tv_apps/browser/js/browser_dialog.js]: this whole createDialog function should create opt with bt1, bt2 being l10nIDs, and then assign them to data-l10n-id in UI.

Comment (zbraniecki) [inline tv_apps/browser/js/browser_dialog.js]: See: https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/localization_code_best_practices#Writing_APIs_that_operate_on_L10nIDs

File: tv_apps/browser/js/search_result.js [hunk 27]
```diff
@@ -0,0 +1,240 @@
...
+  },
+
+  populateResults: function result_populateResults(filter) {
+    var list = this.createList();
+    var engine = SearchUtil.getCurrentEngineName();
+    if( engine ) {
+      var result = this.searchResultTemplate.cloneNode(true);
+      var title = result.childNodes[0].childNodes[0];
+      // title.innerHTML = 'Search by ' + engine;
+      title.innerHTML = _('WB_LT_TIPS_S_SEARCH', {value0:engine});
```
Comment (zbraniecki) [inline tv_apps/browser/js/search_result.js]: this should be:

`navigator.mozL10n.setAttributes(title, 'WB_LT_TIPS_S_SEARCH', {value0: engine});`

File: tv_apps/browser/js/settings.js [hunk 28]
```diff
@@ -1,15 +1,667 @@
...
+    BrowserDialog.createDialog('clear_history', null);
+  },
+
+  handleClearCookieClick: function settings_handleClearCookieClick() {
+    BrowserDialog.createDialog('del_cookie', null);
+  },
+
+  clearCookieFailed: function settings_clearCookieFailed() {
+    // this.settingsBannerMessage.innerHTML = 'Failed to clear cookies and stored data';
+    this.settingsBannerMessage.innerHTML = _('LT_FAILED_');
```
Comment (zbraniecki) [inline tv_apps/browser/js/settings.js]: this should be:

`this.settingsBannerMessage.setAttribute('data-l10n-id', 'LT_FAILED_');

File: tv_apps/browser/js/toolbar.js [hunk 29]
```diff
@@ -1,15 +1,703 @@
...
+   */
+  setSearchEngine: function toolbar_setSearchEngine() {
+    var engineName = SearchUtil.getCurrentEngineName().toLowerCase();
+    // favicon
+    this.searchFavicon.dataset.engine = engineName;
+    // placeholder
+    engineName = engineName.charAt(0).toUpperCase() + engineName.slice(1);
+    this.searchInput.placeholder = engineName;
+    // tooltip
+    this.searchInput.dataset.tips = engineName + ' ' + _('WB_LT_TIPS_SEARCH_BAR');
```
Comment (zbraniecki) [inline tv_apps/browser/js/toolbar.js]: This should be:
`navigator.mozL10n.formatValue('WP_LT_TIPS_SEARCH_BAR', {
  engine: engineName
}).then((value) => {
  this.searchINput.dataset.tips = value;
});`

and `WB_LT_TIPS_SEARCH_BAR` should contain engineName as a variable

File: tv_apps/browser/js/toolbar.js [hunk 30]
```diff
@@ -1,15 +1,703 @@
...
+  },
+
+  /**
+   * Click mode button
+   */
+  clickModeButtonBlock: function toolbar_clickModeButtonBlock() {
+    if(Browser.getCursorPanMode() == 'cursor') {
+      Browser.setCursorPanMode('pan');
+      this.modeButtonTitle.innerHTML = _('WB_LT_PAN_MODE');
+      this.panCursorBannerMessage.innerHTML = _('WB_LT_SWITCH_PAN_MODE');
```
Comment (zbraniecki) [inline tv_apps/browser/js/toolbar.js]: this should just set data-l10n-id on the elements.

File: tv_apps/browser/js/toolbar.js [hunk 31]
```diff
@@ -1,15 +1,703 @@
...
+    this.showPanCursorButton();
+    // Show pan cursor banner
+    this.showPanCursorBanner();
+  },
+  // clear drag mode
+  clearDragMode: function toolbar_clearDragMode(){
+    if(Browser.getCursorPanMode() == 'pan') {
+      Browser.setCursorPanMode('cursor');
+      this.modeButtonTitle.innerHTML = _('WB_LT_CURSOR_MODE');
+      this.panCursorBannerMessage.innerHTML = _('WB_LT_SWITCH_CURSOR_MODE');
```
Comment (zbraniecki) [inline tv_apps/browser/js/toolbar.js]: same here.

File: tv_apps/browser/js/tooltip.js [hunk 32]
```diff
@@ -0,0 +1,186 @@
...
+    var posY = Math.ceil(bounds.bottom);
+    var top = Math.ceil(bounds.top);
+    var tips = null;
+    if(ev.target.dataset.tips != null && ev.target.dataset.tips != undefined) {
+      if(ev.target == Toolbar.searchInput) {
+        tips = ev.target.dataset.tips;
+      } else {
+        tips = _(ev.target.dataset.tips);
+      }
+    }
```
Comment (zbraniecki) [inline tv_apps/browser/js/tooltip.js]: this should be:
`
if(ev.target == Toolbar.searchInput) {
  tips = {raw: ev.target.dataset.tips};
} else {
  tips = ev.target.dataset.tips;
}
`

and then in Tooltip.displaySetting it should be sth like:

`
if (disp.text.raw) {
  element.textContent = disp.text.raw;
} else {
  element.setAttribute('data-l10n-id', disp.text);
}
`

See: https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/localization_code_best_practices#Writing_APIs_that_operate_on_L10nIDs

**Pull request merged**